tartufo/commands/update_signatures.py

```diff
@@ -0,0 +1,196 @@
+    :param config_data: The current tartufo config data
+    :returns: The number of items replaced in config_data
+    """
+    updated = 0
+
+    for old_sig, new_sig in deprecations:
+        targets = functools.partial(lambda o, s: o == s["signature"], old_sig)
+        # Iterate all the deprecations and update them everywhere
+        # they are found in the exclude-signatures section of config
+        for target_signature in filter(targets, config_data["exclude_signatures"]):
+            updated += 1
+            click.echo(f"{updated}) Updating {old_sig!r} -> {new_sig!r}")
+            target_signature["signature"] = new_sig
+
+    return updated
+
+
+def write_updated_signatures(
+    config_path: pathlib.Path, config_data: MutableMapping[str, Any]
+) -> None:
+    """Read the current config file and update it with the new data.
+
+    :param config_path: The path to the tartufo config file
+    :param config_data: The updated config data
+    """
+    with open(str(config_path), "r") as file:
+        result = tomlkit.loads(file.read())
+
+    # Assign the new signatures and write it to the config
+    result["tool"]["tartufo"]["exclude-signatures"] = config_data[  # type: ignore
+        "exclude_signatures"
+    ]
+
+    with open(str(config_path), "w") as file:
+        file.write(tomlkit.dumps(result))
+
+
+@click.command("update-signatures")
+@click.option("--since-commit", help="Only scan from a given commit hash.", hidden=True)
+@click.option(
+    "--max-depth",
+    default=1000000,
+    show_default=True,
+    help="The max commit depth to go back when searching for secrets.",
+    hidden=True,
+)
+@click.option("--branch", help="Specify a branch name to scan only that branch.")
+@click.argument(
+    "repo-path",
+    type=click.Path(exists=True, file_okay=False, resolve_path=True, allow_dash=False),
+)
+@click.option(
+    "--include-submodules/--exclude-submodules",
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help="Controls whether the contents of git submodules are scanned",
+)
+@click.pass_obj
+@click.pass_context
+def main(
+    ctx: click.Context,
+    options: types.GlobalOptions,
+    repo_path: str,
+    since_commit: Optional[str],
+    max_depth: int,
+    branch: Optional[str],
+    include_submodules: bool,
+) -> GitRepoScanner:
+    """Update deprecated signatures for a local repository."""
+    config_path, config_data = load_config_from_path(pathlib.Path(repo_path))
+    if not config_data.get("exclude_signatures"):
+        util.fail(
+            util.style_warning("No signatures found in configuration, exiting..."),
+            ctx,
+            code=0,
+        )
+
+    scanner, stdout, stderr = scan_local_repo(
+        options, repo_path, since_commit, max_depth, branch, include_submodules
+    )
+
+    del stdout  # We are discarding stdout from the scan-local-repo command
+    # Should we print it to the user instead?
+
+    if not scanner:
```

Comment:
```suggestion
    if scanner is None:
```
I believe this is explicit enough to placate mypy without requiring the additional assertion below.

Reply:
Oddly enough, it's not. Both `if not scanner:` and `if scanner is None:` *should* work, but they don't - mypy and pyright both complain.

Is there any reason were actually returning the scanner object from all the commands? Is there any benefit to that?

Reply:
It appears to be because the type checkers cannot infer that the function doesn't continue after the `util.fail` function. 

I can do something like this and it will prevent any assert from ever running and also placates the type checkers:

```py
if not scanner:
    # Explicitly fail if we didn't get a scanner back
    util.fail(util.style_error("Unable to update signatures"), ctx)
    assert False, "Unreachable"
```

Reply:
Oh actually, I am able to get around this if I improve the typehint of `util.fail` as follows:

```py
def fail(msg: str, ctx: click.Context, code: int = 1) -> typing.NoReturn:
    ...
```